internal/character/bailu/talent.go

```diff
@@ -0,0 +1,130 @@
+package bailu
+
+import (
+	"github.com/simimpact/srsim/pkg/engine/event"
+	"github.com/simimpact/srsim/pkg/engine/info"
+	"github.com/simimpact/srsim/pkg/engine/modifier"
+	"github.com/simimpact/srsim/pkg/engine/prop"
+	"github.com/simimpact/srsim/pkg/key"
+	"github.com/simimpact/srsim/pkg/model"
+)
+
+const (
+	invigoration = "invigoration"
+	revive       = "bailu-revive"
+)
+
+// mostly used to track amount of heals left for each chars with invigoration
+type invigStruct struct {
+	healPercent, healFlat float64
+	healsLeft             int
+}
+
+// After an ally with Invigoration is hit, restores the ally's HP
+// for 5.4% of Bailu's Max HP plus 144. This effect can trigger 2 time(s).
+// When an ally receives a killing blow, they will not be knocked down.
+// Bailu immediately heals the ally for 18% of Bailu's Max HP plus 480 HP.
+// This effect can be triggered 1 time per battle
+
+func init() {
+	modifier.Register(invigoration, modifier.Config{
+		Listeners: modifier.Listeners{
+			OnAfterBeingHitAll: healOnBeingHit,
```

Comment:
Invigoration is `OnAfterBeingAttacked`, not `OnAfterBeingHitAll`. This current implementation has invigoration charges being drained and healing after each hit in an attack, and also includes DoT effects, which does not reflect in game behavior.